src/components/listItem/ListItem.test.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { act } from "react";
+import { ListItem } from "./ListItem";
+import { render, screen, waitFor } from "@testing-library/react";
+import { useLists } from "../../contexts/ListsContext";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+
+jest.mock("../../contexts/ListsContext", () => ({ useLists: jest.fn() }));
+
+const mockDelete = jest.fn();
+const mockUpdate = jest.fn();
+
+beforeEach(() => {
+  (useLists as jest.Mock).mockReturnValue({
+    deleteList: mockDelete,
+    updateList: mockUpdate,
+  });
+});
+
+const mockList = {
+  id: "list-1",
+  name: "Groceries",
+  taskCount: 3,
+  tasks: [],
+};
+
+describe("list item components", () => {
+  test("renders list name and task count", () => {
+    render(<ListItem isSelected={false} list={mockList} />);
+    expect(screen.getByText("Groceries")).toBeInTheDocument();
+    expect(screen.getByText("(3)")).toBeInTheDocument();
+  });
+
+  test("enters edit mode on name click", async () => {
```

Comment:
I don't think we need this test, it will be covered when we click and edit the name.